CMakeLists.txt

```diff
@@ -1,48 +1,76 @@
 cmake_minimum_required(VERSION 3.16.3)
-
 project(clio)
 
-if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS 11)
-  message(FATAL_ERROR "GCC 11+ required for building clio")
-endif()
-
-option(BUILD_TESTS "Build tests" TRUE)
-
-option(VERBOSE "Verbose build" TRUE)
-if(VERBOSE)
-  set(CMAKE_VERBOSE_MAKEFILE TRUE)
-  set(FETCHCONTENT_QUIET FALSE CACHE STRING "Verbose FetchContent()")
+# ==================================================== #
+#                       Options                        #
+# ==================================================== #
+option(verbose   "Verbose build"                FALSE) 
+option(tests     "Build tests"                  FALSE)
+option(coverage  "Build test coverage report"   FALSE)
+option(packaging "Create distribution packages" FALSE)
+# ==================================================== #
+
+if(verbose)
+set(CMAKE_VERBOSE_MAKEFILE TRUE)
 endif()
 
-
-if(PACKAGING)
-  add_definitions(-DPKG=1)
+if(packaging)
+add_definitions(-DPKG=1)
 endif()
 
 #c++20 removed std::result_of but boost 1.75 is still using it.
 add_definitions(-DBOOST_ASIO_HAS_STD_INVOKE_RESULT=1)
 
 add_library(clio)
-target_compile_features(clio PUBLIC cxx_std_20)
-target_include_directories(clio PUBLIC src)
 
-include(FetchContent)
-include(ExternalProject)
-include(CMake/settings.cmake)
+# Clio tweaks and checks
+include(CMake/CheckCompiler.cmake)
+include(CMake/Settings.cmake)
 include(CMake/ClioVersion.cmake)
-include(CMake/deps/rippled.cmake)
-include(CMake/deps/libfmt.cmake)
+include(CMake/SourceLocation.cmake)
+
+# Clio deps
+include(CMake/deps/libxrpl.cmake)
 include(CMake/deps/Boost.cmake)
+include(CMake/deps/OpenSSL.cmake)
+include(CMake/deps/Threads.cmake)
+include(CMake/deps/libfmt.cmake)
 include(CMake/deps/cassandra.cmake)
-include(CMake/deps/SourceLocation.cmake)
+
+# TODO: Include directory will be wrong when installed.
+target_include_directories(clio PUBLIC src)
+target_compile_features(clio PUBLIC cxx_std_20)
+
+target_link_libraries(clio
```

Comment:
since we have boost.cmake , why not put these in its own cmake file?

Reply:
John moved everything to the main file which i did not like too much but i did like that you can see everything that is affecting `clio` directly in one place. So i decided to try do it this way. Deps files only make the dependency available (with some specific settings sometimes) but the main cmake file specifies what and how is linked etc.